tests/manage/pv_services/pvc_snapshot/test_encrypted_rbd_pvc_snapshot.py

```diff
@@ -0,0 +1,338 @@
+    ],
+)
+@tier1
+@skipif_ocs_version("<4.8")
+@skipif_ocp_version("<4.8")
+class TestEncryptedRbdBlockPvcSnapshot(ManageTest):
+    """
+    Tests to take snapshots of encrypted RBD Block VolumeMode PVCs
+
+    """
+
+    @pytest.fixture(autouse=True)
+    def setup(self, kv_version, pv_encryption_kms_setup_factory):
+        """ "
+        Setup csi-kms-connection-details configmap
+        """
+
+        log.info("Setting up csi-kms-connection-details configmap")
+        self.vault = pv_encryption_kms_setup_factory(kv_version)
+        log.info("csi-kms-connection-details setup successful")
+
+    def test_encrypted_rbd_block_pvc_snapshot(
+        self,
+        snapshot_factory,
+        snapshot_restore_factory,
+        pod_factory,
+        project_factory,
+        storageclass_factory,
+        multi_pvc_factory,
+        kv_version,
+    ):
+        """
+        Test to take snapshots of encrypted RBD Block VolumeMode PVCs
+
+        """
+        # Create a project
+        proj_obj = project_factory()
+
+        # Create an encryption enabled storageclass for RBD
+        sc_obj = storageclass_factory(
+            interface=constants.CEPHBLOCKPOOL,
+            encrypted=True,
+            encryption_kms_id=self.vault.vault_backend_path,
+        )
+
+        # Create ceph-csi-kms-token in the tenant namespace
+        self.vault.vault_path_token = self.vault.generate_vault_token()
+        self.vault.create_vault_csi_kms_token(namespace=proj_obj.namespace)
+
+        pvc_size = 5
+
+        pvc_objs = multi_pvc_factory(
+            interface=constants.CEPHBLOCKPOOL,
+            project=proj_obj,
+            storageclass=sc_obj,
+            size=pvc_size,
+            access_modes=[
+                f"{constants.ACCESS_MODE_RWX}-Block",
+                f"{constants.ACCESS_MODE_RWO}-Block",
+            ],
+            status=constants.STATUS_BOUND,
+            num_of_pvc=2,
+            wait_each=False,
+        )
+
+        pod_objs = create_pods(
+            pvc_objs,
+            pod_factory,
+            constants.CEPHBLOCKPOOL,
+            pods_for_rwx=1,
+            status=constants.STATUS_RUNNING,
+        )
+        # Verify if the key is created in Vault
+        vol_handles = []
+        for pvc_obj in pvc_objs:
+            pv_obj = pvc_obj.backed_pv_obj
+            vol_handle = pv_obj.get().get("spec").get("csi").get("volumeHandle")
+            vol_handles.append(vol_handle)
+
+            # Check if encryption key is created in Vault
+            if kms.is_key_present_in_path(
+                key=vol_handle, path=self.vault.vault_backend_path
+            ):
+                log.info(f"Vault: Found key for {pvc_obj.name}")
+            else:
+                raise ResourceNotFoundError(f"Vault: Key not found for {pvc_obj.name}")
+
+        log.info(
+            "Check for encrypted device, find initial md5sum value and run IO on all pods"
+        )
```

Comment:
I think the above steps need not be part of the test case. The test case is verifying snapshot and restore of encrypted PVC. So creation of project , storageclass, pvc and pods can be added in the test setup.

Reply:
Done